spec/controllers/users_controller_spec.rb

```diff
@@ -163,29 +188,29 @@
           expect { post :create, params: user_params }.not_to change(User, :count)
 
           expect(response).to have_http_status(:bad_request)
-          expect(JSON.parse(response.body)['errors']).to eq('InviteInvalid')
+          expect(JSON.parse(response.body)['errors']).to eq(Rails.configuration.custom_error_msgs[:invite_token_invalid])
         end
 
         it 'returns an InviteInvalid error if the token is wrong' do
           user_params[:user][:invite_token] = 'fake-token'
           expect { post :create, params: user_params }.not_to change(User, :count)
 
           expect(response).to have_http_status(:bad_request)
-          expect(JSON.parse(response.body)['errors']).to eq('InviteInvalid')
+          expect(JSON.parse(response.body)['errors']).to eq(Rails.configuration.custom_error_msgs[:invite_token_invalid])
         end
       end
 
       context 'approval' do
         before do
           reg_method = instance_double(SettingGetter)
           allow(SettingGetter).to receive(:new).with(setting_name: 'RegistrationMethod', provider: 'greenlight').and_return(reg_method)
-          allow(reg_method).to receive(:call).and_return('approval')
+          allow(reg_method).to receive(:call).and_return(SiteSetting::REGISTRATION_METHODS[:approval])
         end
 
         it 'sets a user to pending when registering' do
           expect { post :create, params: user_params }.to change(User, :count).from(0).to(1)
 
-          expect(User.find_by(email: user_params[:user][:email]).status).to eq('pending')
+          expect(User.find_by(email: user_params[:user][:email])).to be_pending
```

Comment:
User#status is an enum.
ActiveRecord provides useful helpers like (active?, pending?,...) for each enum.
Instead of working with literals we can leverage those helpers,